src/View/SSViewer_BasicIteratorSupport.php

```diff
@@ -57,35 +61,55 @@ public function iteratorProperties($pos, $totalItems)
      *
      * @return bool
      */
-    public function First()
+    public function isFirst()
     {
         return $this->iteratorPos == 0;
     }
 
+    /**
+     * @deprecated 4.7.0 Use isFirst() to avoid clashes with SS_Lists
+     * @return bool
+     */
+    public function First()
+    {
+        Deprecation::notice('4.7.0', 'Use isFirst() to avoid clashes with SS_Lists');
+        return $this->isFirst();
+    }
+
     /**
      * Returns true if this object is the last in a set.
      *
      * @return bool
      */
-    public function Last()
+    public function isLast()
     {
         return $this->iteratorPos == $this->iteratorTotalItems - 1;
     }
 
+    /**
+     * @deprecated 4.7.0 Use isLast() to avoid clashes with SS_Lists
+     * @return bool
+     */
+    public function Last()
+    {
+        Deprecation::notice('4.7.0', 'Use isLast() to avoid clashes with SS_Lists');
```

Comment:
Let's start throwing deprecation at 5.0. I don't think we should force everyone using Silverstripe 4 to rewrite all their templates for the sake of clarity.

Reply:
Yeah 5 makes sense, will change this also.